client/extensions/woocommerce/woocommerce-services/state/shipping-label/actions.js

```diff
@@ -104,9 +104,21 @@ const waitForAllPromises = ( promises ) => {
 	return Promise.all( promises.map( ( p ) => p.catch( ( e ) => e ) ) );
 };
 
-const getNextErroneousStep = ( form, errors, currentStep ) => {
-	const firstStepToCheck = FORM_STEPS[ FORM_STEPS.indexOf( currentStep ) + 1 ];
-	switch ( firstStepToCheck ) {
+/**
+ * Recursively checks the form for errors and returns a step with an error in it or null
+ * @param {Object} form the form
+ * @param {Object} errors error tree
+ * @param {Number} currentStepIndex (optional) index of the step where the check should start, defaults to 0
+ *
+ * @returns {String} erroneous step name or null
+ */
+const getNextErroneousStep = ( form, errors, currentStepIndex = 0 ) => {
```

Comment:
Since `getNextErroneousStep` always checks the whole form, the `switch` & recursive approach doesn't make sense anymore. It would be simpler:

```js
const getNextErroneousStep = ( form, errors ) => {
  if ( errorsInOrigin ) return 'origin';
  if ( errorsInDestination ) return 'destination';
  ...
  return null;
}
```

As an aside, `getNextErroneousStep` should now be called `getFirstErroneousStep`.

Reply:
It would be great if you take a look at the other suggestion, the recursive approach to this logic is over-complicated now that the current step doesn't matter.